multimap/src/main/java/org/infinispan/multimap/impl/ListBucket.java

```diff
@@ -250,6 +250,29 @@ public ListBucket<V> insert(boolean before, V pivot, V element) {
       return new ListBucket<>(newValues);
    }
 
+   public long remove(long count, V element) {
+      Iterator<V> ite;
+
+      if (count < 0) {
+         ite = values.descendingIterator();
+      } else {
+         ite = values.iterator();
+      }
+
+      long maxRemovalsCount = count == 0 ?  values.size(): Math.abs(count);
+      long removedElements = 0;
+      while(ite.hasNext() && maxRemovalsCount > 0) {
+         V next = ite.next();
+         if (Objects.deepEquals(next, element)) {
+            maxRemovalsCount--;
+            removedElements++;
+            ite.remove();
+         }
+      }
```

Comment:
can be optimized, something like this
```suggestion
      while(ite.hasNext()) {
         V next = ite.next();
         if (Objects.deepEquals(next, element)) {
            ite.remove();
            if (++removedElements==maxRemovalsCount) {
              break;
            }
         }
```